### ISSUE TITLE
[RFE]: Restart fail2ban without altering the state

### ISSUE THREAD

**Issue by brianjmurrell** [fail2ban/fail2ban#3164]
#### Feature request type
I simply want to restart a running instance of fail2ban 0.11.2 without the rigmarole of flushing/unbanning/rebanning/etc. 1000s of hosts.

I want fail2ban to trust that the state of whatever is being used to block hosts does not change between it's stop and start (i.e. restart) and just have it stop without unbanning and resume operations on start with the pre-existing state and try to re-instate bans.

#### Description
For some banning mechanisms, it can be relatively expensive to ban/unban.  It's such that during the normal operations of fail2ban, the expense is affordable, but the expense of unbanning 1000s of hosts only to re-ban them again is just too much, and frankly silly.  So some mechanism to just restart fail2ban (i.e. because of an upgrade) without it going through a massive unban/ban would be desirable.

#### Considered alternatives
None

#### Any additional information
None

**Comment by brianjmurrell** [comment]
I would have thought that setting:

```
norestored = 1
actionflush = true
```

in my action would achieve this but I still get a ban storm on [re-]start.

**Comment by sebres** [comment]
Dup of #3085, #3038 et al.

> but I still get a ban storm on [re-]start.

Well, the setting `actionflush` would work, but if you initially started fail2ban without that, it'd work only after restart, because changing of configuration does not update the run-time values unless you used `reload`. So there are 2 ways to achieve that:
1. either set that options to the action and then:
```bash
fail2ban-client reload && fail2ban-client restart
```
2. or change run-time values directly (per jail) and then do restart:
```bash
#fail2ban-client set "$jail" action "$action" norestored 1
fail2ban-client set "$jail" action "$action" actionflush true
fail2ban-client restart
```
So for the unban by stop (e. g. `actionflush`)... but "a ban storm" after restart is not explainable, because if `norestored = 1` is set by the action, it would definitely not ban restored tickets (all active tickets fail2ban stored in database).
Sure you'll still see a lot of `Restore Ban ...` in log, but the `actionban` should be definitely not called for this.
But the fail2ban still needs to restore all active tickets (to know which entries need to be unbanned if they get expired later if end of ban will be reached).

If the action is still invoked on your side on restart, I guess something other happens there... For example you has very large `findtime` and fail2ban database doesn't contain last position in log, so it begins to check the log/journal starting from `now - findtime` and ban every evildoer making `maxretry` attempts - this all not restored tickets and `norestored` doesn't affect them.
Or you probably set  `norestored` not at correct place - note that this are action (and not jail) parameters, so you have to set it either in the definition section of action (e.g. in `action.d/my-action.local`) or in the `jail.local` but then like this:
```ini
banaction = my-action[norestored=1, actionflush=true]
```

Just note that the setting `norestored=1` and `actionflush=true` have an issue with potentially inconsistent state for some ticket.
Imagine you have some tickets which must be normally unbanned at `12:15:00` exactly. If you stop (restart) fail2ban at `12:14:58`  and after start it begins a restore process for this jail at `12:15:01` this tickets become obsolete to the time of ban (restore), so fail2ban avoid to manage such tickets, but because you don't flushed them by stop (restart), they will remain banned for infinite time (since fail2ban doesn't keep them under its control at all).
This "regression" cannot be managed gracefully, so it is rather to see as a workaround, I hope it is clear for everyone using this "solution".
Proper solution can be implemented if bulk banning (#3088) becomes ready. But it will imply that your action is fundamentally able to do such mass-banning.
